crates/iceberg-catalog/src/service/storage/error.rs

```diff
@@ -150,6 +159,9 @@ impl From<TableConfigError> for IcebergErrorResponse {
             e @ TableConfigError::Misconfiguration(_) => {
                 ErrorModel::bad_request(e.to_string(), "Misconfiguration", Some(Box::new(e))).into()
             }
+            e @ TableConfigError::StsError(_, _) => {
+                ErrorModel::internal(e.to_string(), "StsError", Some(Box::new(e))).into()
```

Comment:
Also here - bad_request?

Reply:
When getting table config it's more on the internal error, maybe failed dependency, there's nothing wrong with the GET table request we received, we either have a bad state internally or some external cloud provider config has changed